test/python/pulse/compiler_passes/test_schedule.py

```diff
@@ -0,0 +1,422 @@
+        self.assertEqual(ir_example.time_table[3], 0)
+        self.assertEqual(ir_example.time_table[4], 200)
+
+    def test_recursion_to_sub_blocks(self):
+        """test that scheduling is recursively applied to sub blocks which are first in the order"""
+
+        sub_block = SequenceIR(AlignRight())
+        sub_block.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+
+        ir_example = SequenceIR(AlignRight())
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(0))))
+        ir_example.append(sub_block)
+
+        ir_example = self._pm.run(ir_example)
+        sub_block = ir_example.elements()[1]
+        # Note that sub blocks are oblivious to their relative timing
+        self.assertEqual(sub_block.initial_time(), 0)
+        self.assertEqual(sub_block.final_time(), 100)
+        self.assertEqual(sub_block.scheduled_elements()[0][0], 0)
+
+    def test_with_parallel_sub_block(self):
+        """test with a sub block which doesn't depend on previous instructions"""
+
+        sub_block = SequenceIR(AlignRight())
+        sub_block.append(Play(Constant(200, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+
+        ir_example = SequenceIR(AlignRight())
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(0))))
+        ir_example.append(sub_block)
+
+        ir_example = self._pm.run(ir_example)
+        self.assertEqual(ir_example.initial_time(), 0)
+        self.assertEqual(ir_example.final_time(), 200)
+        self.assertEqual(ir_example._time_table[2], 100)
+        self.assertEqual(ir_example._time_table[3], 0)
+
+    def test_with_sequential_sub_block_with_more_dependencies(self):
+        """test with a sub block which depends on a several previous instruction"""
+
+        sub_block = SequenceIR(AlignRight())
+        sub_block.append(Delay(100, target=Qubit(0)))
+
+        ir_example = SequenceIR(AlignRight())
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(0))))
+        ir_example.append(Play(Constant(200, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(1), QubitFrame(1))))
+        ir_example.append(sub_block)
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(1), QubitFrame(1))))
+
+        ir_example = self._pm.run(ir_example)
+
+        self.assertEqual(ir_example.initial_time(), 0)
+        self.assertEqual(ir_example.final_time(), 300)
+        self.assertEqual(ir_example.time_table[2], 100)
+        self.assertEqual(ir_example.time_table[3], 0)
+        self.assertEqual(ir_example.time_table[4], 100)
+        self.assertEqual(ir_example.time_table[5], 200)
+        self.assertEqual(ir_example.time_table[6], 200)
+
+
+class TestScheduleAlignSequential(SchedulingTestCase):
+    """Test Schedule pass with AlignSequential"""
```

Comment:
Maybe we can comment scheduling functionality is the same with the left one (both ASAP) and this just checks dependency on the sequence pass?

Reply:
I'll add some words to the doc string.